controller/api/destination/watcher/ip_watcher.go

```diff
@@ -145,12 +168,45 @@ func (iw *IPWatcher) GetSvcID(clusterIP string) (*ServiceID, error) {
 	return service, nil
 }
 
-// GetPod returns the pod that corresponds to an IP address if one exists.
-func (iw *IPWatcher) GetPod(podIP string) (*corev1.Pod, error) {
-	objs, err := iw.k8sAPI.Pod().Informer().GetIndexer().ByIndex(podIPIndex, podIP)
+// GetPod returns a pod that maps to the given IP address. The pod can either
+// be in the host network or the pod network. If the pod is in the host
+// network, then it must have a container port that exposes `port` as a host
+// port.
+func (iw *IPWatcher) GetPod(podIP string, port uint32) (*corev1.Pod, error) {
+	// First we check if the address maps to a pod in the host network.
+	addr := fmt.Sprintf("%s:%d", podIP, port)
+	hostIPPods, err := iw.getIndexedPods(hostIPIndex, addr)
+	if err != nil {
+		return nil, status.Error(codes.Unknown, err.Error())
```

Comment:
When can this actually error? Should we continue attempting the pod-network lookup when it does?

Reply:
It errors only if the k8s API errors when looking up the endpoint in the index. That would mean the API is already in a bad state so I don't think purposely continuing is really the right thing to do—even if it is logged as an error here.